File: app/src/main/java/com/txt_nifty/sketch/flmml/rep/MacroArgument.java
package com.txt_nifty.sketch.flmml.rep;

public class MacroArgument {
    public int index;
    public String id;
    public String code;
    public MacroArgument[] args = new MacroArgument[0];

    public MacroArgument(String id, int index) {
        this.index = index;
        this.id = id;
    }

    public MacroArgument(String id, String code, MacroArgument[] args) {
        this.id = id;
        this.code = code;
        if (args != null)
            this.args = args;
    }
}

File: app/src/main/java/com/txt_nifty/sketch/flmml/rep/MacroArgument.kt
package com.txt_nifty.sketch.flmml.rep

class MacroArgument {
    var index = 0
    var id: String
    var code: String? = null
    var args = emptyArray<MacroArgument>()

    constructor(id: String, index: Int) {
        this.index = index
        this.id = id
    }

    constructor(id: String, code: String?, args: Array<MacroArgument>?) {
        this.id = id
        this.code = code
        if (args != null) this.args = args
    }
}